fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/impl/transfer/transaction/TransactionManagerImpl.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.transfer.transaction;
+
+import java.util.ArrayList;
+
+import net.fabricmc.fabric.api.transfer.v1.transaction.Transaction;
+
+public class TransactionManagerImpl {
```

Comment:
a useful debug feature is to record the stacktrace from where the Transaction was created, if someone forgets to close their transaction, the exception can help narrow down who's responsible.
https://github.com/Astrarre/Astrarre/blob/4913a4cb9fa6e88183f31e5f2e5f77fb4f485bd4/astrarre-transfer-v0/src/main/java/io/github/astrarre/transfer/v0/api/transaction/Transaction.java#L38

Reply:
Hmmm acquiring the stacktrace is quite an expensive operation, I certainly don't want this to happen by default. The risk that someone forgets to close the transaction is abysmal assuming they are always used within a `try`-with-resources block?

Reply:
> Hmmm acquiring the stacktrace is quite an expensive operation, I certainly don't want this to happen by default.

ye atm I only do it in dev or with a system variable